contracts/facets/ThirdPartyFacet.sol

```diff
@@ -91,117 +89,124 @@ contract ThirdPartyFacet {
+
+        // the sarcophagus is compromised if k or more archaeologists have been accused successfully on this call
+        bool isSarcophagusCompromised = accusalCount >= sarco.minShards;
+
+        // if the current call hasn't resulted in at least k archaeologists being accused
+        // check if total number of historical accusals on sarcophagus is greater than k
+        uint historicalAccusals = 0;
+        if (!isSarcophagusCompromised) {
+            for (uint256 i = 0; i < archaeologistAddresses.length; i++) {
+                if (s.sarcophagusArchaeologists[sarcoId][archaeologistAddresses[i]].accused) {
+                    historicalAccusals++;
                 }
             }
+            // the sarcophagus is compromised if k or more archaeologists have been accused over the lifetime of the sarcophagus
+            if (historicalAccusals >= sarco.minShards) {
+                isSarcophagusCompromised = true;
+            }
+        }
 
-            // If this arch address wasn't in the accused list, free it from its curse
-            if (isUnaccused) {
-                // There are technically no rewards here, since the sarcophagus
-                // has been compromised, so here this effectively merely resets
-                // the state of the non-malicious archaeologists, as if they never
-                // bonded to this sarcophagus in the first place.
-                //
-                // Of course, whatever rewards they might have gained in previous
-                // rewraps remains theirs.
-                LibBonds.freeArchaeologist(sarcoId, bondedArchaeologists[i]);
+        // if k or more archaeologists have been accused over the lifetime of the sarcophagus, funds should
+        // be returned to the remaining well behaved archaeologists
+        if (isSarcophagusCompromised) {
+            // update the sarcophagus state to Accused, the outer key is compromised
+            sarco.state = LibTypes.SarcophagusState.Accused;
+
+            // iterate through all archaeologist addresses on the sarcophagus
+            for (uint256 i = 0; i < archaeologistAddresses.length; i++) {
+                // if the archaeologist has never been accused, release their locked bond back to them
+                if (!s.sarcophagusArchaeologists[sarcoId][archaeologistAddresses[i]].accused) {
+                     LibBonds.freeArchaeologist(sarcoId, archaeologistAddresses[i]);
+                }
             }
         }
 
+        // refund the diggingFees allocated by the embalmer to the accused archaeologists
+        // split the total bond being slashed between the embalmer and the payment address
         (
             uint256 accuserBondReward,
             uint256 embalmerBondReward
         ) = _distributeLoot(
                 paymentAddress,
                 sarco,
-                totalCursedBond,
-                diggingFeesToBeDistributed
+                totalDiggingFees,
+                totalDiggingFees
```

Comment:
Could we not now remove one of these params?

Reply:
haha definitely